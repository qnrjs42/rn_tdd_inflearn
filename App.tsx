import React from 'react';
import {SafeAreaView, Text, View} from 'react-native';

const App = () => {
  return (
    <SafeAreaView>
      <View testID="welcome">
        <Text>Welcome to React Native!</Text>
      </View>
    </SafeAreaView>
  );
};

export default App;
