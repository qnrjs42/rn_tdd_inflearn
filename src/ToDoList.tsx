import {View, Text} from 'react-native';
import React, {FC} from 'react';

const ToDoList: FC = () => {
  return (
    <View>
      <Text>ToDoList</Text>
    </View>
  );
};

export default ToDoList;
