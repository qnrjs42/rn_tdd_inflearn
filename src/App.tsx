import React from 'react';
import {SafeAreaView, Text, View} from 'react-native';

import AddToDo from './AddToDo';
import ToDoList from './ToDoList';

const App = () => {
  return (
    <SafeAreaView>
      <View testID="welcome">
        <Text>ToDo TDD</Text>
        <AddToDo />
        <ToDoList />
      </View>
    </SafeAreaView>
  );
};

export default App;
