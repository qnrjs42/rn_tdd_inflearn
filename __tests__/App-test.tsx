import React from 'react';
import {render, screen} from '@testing-library/react-native';

import App from '../src/App';
import AddToDO from '../src/AddToDo';
import ToDoList from '../src/ToDoList';

describe('App', () => {
  beforeAll(() => {
    render(<App />);
  });

  test('is Text visible?', async () => {
    const text = screen.getByText('ToDo TDD');
    expect(text).toBeTruthy();
  });

  test('is <AddToDO> visible?', async () => {
    const {toJSON} = render(<AddToDO />);
    expect(toJSON()).toMatchSnapshot();
  });

  test('is <ToDoList> visible?', async () => {
    const {toJSON} = render(<ToDoList />);
    expect(toJSON()).toMatchSnapshot();
  });
});

// describe('Jest', () => {
//   it('is it working?', () => {
//     const a = 1;
//     expect(a + 1).toBe(2);
//   });
// });

// test('examples of some things', async () => {
//   render(<Text>Hello</Text>);

//   expect(screen.toJSON()).toMatchSnapshot();
// });
