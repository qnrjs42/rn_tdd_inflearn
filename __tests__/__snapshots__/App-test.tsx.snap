// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App is <AddToDO> visible? 1`] = `
<View>
  <Text>
    AddToDo
  </Text>
</View>
`;

exports[`App is <ToDoList> visible? 1`] = `
<View>
  <Text>
    ToDoList
  </Text>
</View>
`;
